### PR TITLE
Update command fccc

### LINKED ISSUE #56: FCC Challenge Function
With the release of the new FCC curriculum and subdomain, it is fairly safe to assume our scraper based FCC challenge function is now broken.

### PR DESCRIPTION
**Pull request by ghost**
fccc

# Resolves

Resolves #56 

# Changes

* src/bot.js

I had to make a decision about when to make fetch requests to get fcc curriculum data from FCC's Curriculum repo. Fetching this data every time a request is made is not practical. I chose instead that a sane option would be to fetch all the data when the bot starts, as opposed to when the bot installs.

* src/commands/scrapers/data
* src/commands/scrapers/data/fcc
* src/commands/scrapers/data/fcc/githubEndpoints.json

These folders were created to help organize fetch endpoints and response data. githubEndpoints.json was created, because I was unable to automate the process of determining FCC's curriculum structure programmatically.

* .gitignore

added src/commands/scrapers/data/fcc/challengeData.json - the response data from fetching githubEndpoints.json

# Testing

Manual testing was performed and there appears to be no issues with any of the commands or core bot code.

### CODE REVIEW

**Comment by rricheydev** [comment]
Updated PR to fix/support the URL link pointing to the challenge's location on learn.freecodecamp.org.

**Comment by joker314** [comment]
This is a really good pull request. Briefly scanning through the code, I figure we could replace a few more elements.

- `` could be transformed into a markdown image `alt` (or link if images not supported)
- `<a>` could be transformed into a markdown link `[text](url)`
- `<b>`, `<strong>` could be transformed into a markdown bold `**text**`
- `<i>`, `<em>` could be transformed into a markdown italic `*text*`

Thoughts?

The `` replacement would have to be done before stripping self-closing tags. In addition, would we want to replace `<hr>` with a few underscores (or even an image of a horizontal grey line?)

**Comment by rricheydev** [comment]
I consider handling img and a tags to be out-of-scope/unnecessary. I have not seen any evidence of img or a tags being used in challenge descriptions. They appear to be used exclusively as content for the built-in editor. When appearing in a description, they are HTML encoded as `&lt;...&gt;` which then become displayed as html tags.

br, b, and i tags have been handled.

**Comment by joker314** [comment]
Alright, could you add
- `<dfn>` `<em>` - italics
- `<strong>` bold

Image tags are being phased out for screen readers. Links, however, remind me of another thing (perhaps out of scope of this PR, but possibly for a follow-up), I believe at the footer of challenges is a list of links to helpful resources. This could be added to the search functionality?

<hr>

Also, would it be useful to add a `url` parameter to the embed. This makes the title clickable, and will open up the challenge page?

**Comment by rricheydev** [comment]
v2 added back in support for URL clickable titles that send users to the FCC challenge page. It's possible that some challenges do have links in the description, but not all - perhaps a good follow-up issue if needed.

Other changes made as requested. Thanks for the feedback 👍

File: src/commands/scrapers/FCCCommand.js [hunk 1]
```diff
@@ -72,8 +72,8 @@ function removeHTMLTags(textArray) {
     .replace(/<code>|<\/code>/g, '`')
     .replace(/<blockquote>|<\/blockquote>/g, '```')    
     .replace(/<b>|<\/b>|<strong>|<\/strong>/g, '**')
-    .replace(/<i>|<\/i>/g, '*')
-    .replace(/<br>/g, '\n')
+    .replace(/<i>|<\/i>|<em>|<\/em>|<dfn>|<\/dfn>/g, '*')
+    .replace(/<br>/g, '\n')
```
Comment (joker314) [inline src/commands/scrapers/FCCCommand.js]: Could you remove this trailing whitespace?

Comment (rricheydev) [reply src/commands/scrapers/FCCCommand.js]: There's trailing white space all over the place. Is there something special about that particular trailing white space?

Comment (joker314) [reply src/commands/scrapers/FCCCommand.js]: Yes -- that one happened to catch my eye 😛.

At some point in the distant feature, we should clean the whitespace up. However, I guess we should do it in one go in another PR, when we add linting. LGTM

**Comment by rricheydev** [comment]
squashed 5 commits into 1

**Pull request merged**